flight/Libraries/math/lqg.c

```diff
@@ -46,8 +46,16 @@
 
 */
 
-#define RTKF_SOLUTION_LIMIT			150
-#define LQR_SOLUTION_LIMIT			1000
+#define SOLVER_MIN					100
+#define SOLVER_MAX					5000
+
+/*
+	Applies to torque and bias parts.
+
+	Beta 7.5 and Tau 50ms solves RTKF in 3077 iterations and the LQR is 1267 iterations.
+	Beta 10.5 and Tau 15ms solves RTKF in 298 iterations and the LQR in 156 iterations.
+*/
+#define SOLVER_DELTA_EPSILON		0.000000001f
```

Comment:
Are you sure this is safe / can always be reached?  Only on really small floating point values can you get a (nonzero) difference this small.

Reply:
Seems to work out for all the random values I tossed at it (from beta 6 tau 150ms to beta 12 tau 5ms), assuming non-fastmath IEEE-754 works the same between ARM and x86.

Reply:
Basically float can be assumed to have 6 sigfigs; this number is 10^-9, so it means that this epsilon "works" on values that are less than 10^-3; otherwise it can be expected to do weird things.